src/main/scala/chisel3/util/experimental/BoringUtils.scala

```diff
@@ -98,7 +99,7 @@ class BoringUtilsException(message: String) extends Exception(message)
   */
 object BoringUtils {
   /* A global namespace for boring ids */
-  private val namespace: SyncVar[Namespace] = new SyncVar
+  private val namespace: LinkedBlockingDeque[Namespace] = new LinkedBlockingDeque[Namespace](1)
```

Comment:
This should probably use the new Builder ContextCache: https://github.com/chipsalliance/chisel3/pull/3009

Reply:
I agree, I reverted this change so as to unblock the rest of these changes and we can fix that separately.